Build failure for `samples/net/gptp` on `sam_e70_xplained/same70q21`

### Describe the bug

Commit cbd3ebcc2d43998fa7de8675bec2ba65ed724f77 (#91040) introduces a compiler build error for the `samples/net/gptp` sample.

### Regression

- [x] This is a regression.

### Steps to reproduce

Run

```shell
west build -b sam_e70_xplained/same70q21 samples/net/gptp
```

### Relevant log output

```shell
/home/pdgendt/zephyrproject/zephyr/drivers/ethernet/eth_sam_gmac.c:180:40: warning: 'tx_frame_list0_que' defined but not used [-Wunused-variable]
  180 |                 static struct net_pkt *tx_frame_list##n##_que[NET_PKT_PER_QUE *         \
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/drivers/ethernet/eth_sam_gmac.c
+++ b/drivers/ethernet/eth_sam_gmac.c
@@ -2182,10 +2182,10 @@ static const struct ethernet_api eth_api = {
 		},
 
 #if defined(CONFIG_PTP_CLOCK_SAM_GMAC)
-#define BUF_TX_FRAME_LIST_QUE (uint32_t *)&tx_frame_list##n##_que[x * NET_PKT_PER_QUE]
+#define BUF_TX_FRAME_LIST_QUE(n, x) (uint32_t *)&tx_frame_list##n##_que[x * NET_PKT_PER_QUE]
 #define DEFN_TX_FRAME_LIST(n, x)							\
 		.tx_frames = {								\
-			.buf = BUF_TX_FRAME_LIST_QUE,					\
+			.buf = BUF_TX_FRAME_LIST_QUE(n, x),				\
 			.len = NET_PKT_PER_QUE,						\
 		},
 #else /* CONFIG_PTP_CLOCK_SAM_GMAC */

